### PR TITLE
fix: support OTP2 SCOOTER mode

### LINKED ISSUE #379: OTP2 Scooter Legs Not Rendered
OTP2 scooter legs use a different designation, causing them not to be picked up by transitive. This is possibly an issue that needs fixing inside `core-utils`

Screen Shot 2022-04-18 at 2 36 33 PM

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
Closes #379. OTP2 uses `SCOOTER` instead of `MICROMOBILITY`, and we need to account for this.

### CODE REVIEW

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Dang this thing really appeared everywhere.

File: packages/core-utils/src/map.js [hunk 1]
```diff
@@ -106,11 +106,13 @@ export function itineraryToTransitive(
       leg.mode === "WALK" ||
       leg.mode === "BICYCLE" ||
       leg.mode === "CAR" ||
-      leg.mode === "MICROMOBILITY"
+      leg.mode === "MICROMOBILITY" ||
```
Comment (binh-dam-ibigroup) [inline packages/core-utils/src/map.js:109]: We could probably use `isAccessMode` for the entire `if` condition.

File: packages/itinerary-body/src/AccessLegBody/rented-vehicle-subheader.tsx [hunk 2]
```diff
@@ -27,12 +27,12 @@ interface Props {
...
 export default function RentedVehicleSubheader({ config, leg }: Props) {
   const configCompanies = config.companies || [];
-  const { from, mode } = leg;
+  const { from, mode, rentedBike, walkingBike } = leg;
   const { name: legName, networks, vertexType: modeType } = from;
 
   // Sometimes rented vehicles can be walked over things like stairs or other
   // ways that forbid the main mode of travel.
-  if (mode === "WALK") {
+  if (mode === "WALK" || walkingBike === true) {
```
Comment (binh-dam-ibigroup) [inline packages/itinerary-body/src/AccessLegBody/rented-vehicle-subheader.tsx:35]: Can we just write `|| walkingBike`?

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM

File: packages/core-utils/src/map.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply packages/core-utils/src/map.js]: Good call thanks! Refactored in e351b70d8907551a03043bf49d583317231644d2

**Pull request merged**